### ISSUE TITLE
LightningModule attribute as a list of torchmetrics Metric gives error with DDP

### ISSUE THREAD

**Issue by shenoynikhil** [Lightning-AI/torchmetrics#2707]
## 📚 Documentation
I have a custom metric implemented using the torchmetrics Metric. For one of my applications, if I instantiate an attribute of my LightningModule class as a list of my metric, there are issues during the sync/gather operation during `metric.compute()`. But doing so with a `torch.nn.ModuleList` works,
```
self.metrics = torch.nn.ModuleList([MyMetric() for _ in range(2)]) # works
self.metrics = [MyMetric() for _ in range(2)] # does not work :(
```
I am not sure if this is common knowledge, but I found it a bit surprising. Again, unsure if its a bug, this could be highlighted in the documentation.

Thanks!

**Comment by SkafteNicki** [comment]
Hi @shenoynikhil, thanks for opening this issue.
I would say that this particular problem is fairly well documented in the documentation:
https://lightning.ai/docs/torchmetrics/stable/pages/overview.html#metrics-and-devices
So I am going to close this issue.
